src/lsm/table_memory.zig

```diff
@@ -29,8 +31,11 @@ pub fn TableMemoryType(comptime Table: type) type {
                 suffix_offset: u32 = 0,
             },
             immutable: struct {
-                /// An empty table has nothing to flush
+                /// An empty table has nothing to flush.
                 flushed: bool = true,
+                /// This field is only used for assertions, to verify that we don't absorb the
+                /// mutable table immediately prior to checkpoint.
+                absorbed: bool = false,
```

Comment:
I wonder if `flushed` & `absorbed` could be merged into a single state enum, for tighter asserts? 

Previously, we used `flushed` to assert that we didn't forget to flush, but now we implicitly make decision that if not flushed, than we should absorb. 

Though, to be fair, I can't immediately come up with good names for the enum variants!

Reply:
These flags are not actually orthogonal. (They _usually_ are, but not strictly.)

At the end of `TableMemory.absorb()`, when we set `absorbed` to `true`:

```zig
            table_immutable.mutability = .{ .immutable = .{
                .flushed = table_immutable.value_context.count == 0,
                .absorbed = true,
                .snapshot_min = snapshot_min,
            } };
```

...it is possible that we set `flushed` to `true` at the same time, if all of the keys in the mutable table annihilate all of the keys in the immutable table.